### PR TITLE
Python3 compat

### PR DESCRIPTION
**Pull request by vdumoulin**

### CODE REVIEW

**Comment by vdumoulin** [comment]
@carriepl @abergeron @lamblin @nouiz can anyone of you have a quick look?

File: convnets/logistic_sgd.py [hunk 1]
```diff
@@ -428,8 +428,8 @@ def sgd_optimization_mnist(learning_rate=0.13, n_epochs=1000,
         )
         % (best_validation_loss * 100., test_score * 100.)
     )
-    print 'The code run for %d epochs, with %f epochs/sec' % (
-        epoch, 1. * epoch / (end_time - start_time))
+    print('The code run for %d epochs, with %f epochs/sec' % (
+        epoch, 1. * epoch / (end_time - start_time)))
 
     print >> sys.stderr, ('The code ran for %.1fs' % ((end_time - start_time)))
```
Comment (lamblin) [inline convnets/logistic_sgd.py]: You probably have to update that one as well.

**Comment by lamblin** [comment]
All of these make sense, so we can merge. There may still be other missing conversions needed.

File: convnets/logistic_sgd.py (hunk 1 above)
Comment (vdumoulin) [inline convnets/logistic_sgd.py]: Surprisingly, I had no problem running this line with Python 3.

Comment (bartvm) [inline convnets/logistic_sgd.py]: That's very strange, given that it is even an example of syntax not
supported in Python 3 anymore:
https://docs.python.org/3.0/whatsnew/3.0.html#print-is-a-function :/

Comment (vdumoulin) [inline convnets/logistic_sgd.py]: Nevermind, I was running the notebooks and making sure that nothing crashed, but this file isn't being used by any notebook. You're right, this statement causes a syntax error to be raised in Python 3.

**Comment by vdumoulin** [comment]
I replaced the `print >>` statements with something python 2/3 compatible and did some additional adaptations.

**Comment by abergeron** [comment]
Looks good for the convnet part.

**Pull request merged**